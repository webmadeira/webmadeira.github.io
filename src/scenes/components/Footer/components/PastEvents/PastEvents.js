import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { H3, Caption } from '../../../../../components/typography'

const WebConferencesList = styled.ul`
  list-style-type: none;
  text-align: left;
  margin-top: 15px;
`

const WebConferencesListItem = styled.li`
  margin-bottom: 8px;
`

const WebConferenceLink = styled.a`
  color: ${props => props.theme.color('neutral', 'bright')};
`

const getPastEventsItems = events => events.map(event => (
  <WebConferencesListItem key={event.id}>
    <WebConferenceLink href={event.url} rel="noopener" target="_blank">
      <Caption>{event.name}</Caption>
    </WebConferenceLink>
  </WebConferencesListItem>
))

const PastEvents = ({ pastEvents }) => (
  <div>
    <H3>PAST EVENTS</H3>
    <WebConferencesList>
      {getPastEventsItems(pastEvents)}
    </WebConferencesList>
  </div>
)

PastEvents.propTypes = {
  pastEvents: PropTypes.PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      url: PropTypes.string,
    })).isRequired,
}

export default PastEvents
