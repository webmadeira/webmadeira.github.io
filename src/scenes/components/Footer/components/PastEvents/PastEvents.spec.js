import React from 'react'
import { shallow } from 'enzyme'

import PastEvents from './PastEvents'

describe('PastEvents component', () => {
  it('must render properly', () => {
    const pastEvents = [
      {
        id: 0,
        name: 'Test event',
        url: 'http://test.url.com/',
      },
    ]

    const component = shallow(<PastEvents pastEvents={pastEvents} />)

    expect(component).toMatchSnapshot()
  })
})
