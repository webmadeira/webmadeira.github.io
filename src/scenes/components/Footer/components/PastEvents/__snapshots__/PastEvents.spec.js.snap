// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastEvents component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PastEvents
    pastEvents={
      Array [
        Object {
          "id": 0,
          "name": "Test event",
          "url": "http://test.url.com/",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.h3>
          PAST EVENTS
        </styled.h3>,
        <styled.ul>
          <styled.li>
            <styled.a
              href="http://test.url.com/"
              rel="noopener"
              target="_blank"
            >
              <styled.span>
                Test event
              </styled.span>
            </styled.a>
          </styled.li>
        </styled.ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "PAST EVENTS",
        },
        "ref": null,
        "rendered": "PAST EVENTS",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.li>
              <styled.a
                href="http://test.url.com/"
                rel="noopener"
                target="_blank"
              >
                <styled.span>
                  Test event
                </styled.span>
              </styled.a>
            </styled.li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": <styled.a
                href="http://test.url.com/"
                rel="noopener"
                target="_blank"
              >
                <styled.span>
                  Test event
                </styled.span>
              </styled.a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.span>
                  Test event
                </styled.span>,
                "href": "http://test.url.com/",
                "rel": "noopener",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Test event",
                },
                "ref": null,
                "rendered": "Test event",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.h3>
            PAST EVENTS
          </styled.h3>,
          <styled.ul>
            <styled.li>
              <styled.a
                href="http://test.url.com/"
                rel="noopener"
                target="_blank"
              >
                <styled.span>
                  Test event
                </styled.span>
              </styled.a>
            </styled.li>
          </styled.ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "PAST EVENTS",
          },
          "ref": null,
          "rendered": "PAST EVENTS",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.li>
                <styled.a
                  href="http://test.url.com/"
                  rel="noopener"
                  target="_blank"
                >
                  <styled.span>
                    Test event
                  </styled.span>
                </styled.a>
              </styled.li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": <styled.a
                  href="http://test.url.com/"
                  rel="noopener"
                  target="_blank"
                >
                  <styled.span>
                    Test event
                  </styled.span>
                </styled.a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <styled.span>
                    Test event
                  </styled.span>,
                  "href": "http://test.url.com/",
                  "rel": "noopener",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Test event",
                  },
                  "ref": null,
                  "rendered": "Test event",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
